src/main/kotlin/com/idan/pokemon_hub/service/PokemonService.kt

```diff
@@ -0,0 +1,43 @@
+package com.idan.pokemon_hub.service
+
+import com.idan.pokemon_hub.exception.InvalidFieldException
+import com.idan.pokemon_hub.exception.PokemonNotFoundException
+import com.idan.pokemon_hub.model.Pokemon
+import com.idan.pokemon_hub.repository.PokemonRepository
+import org.springframework.stereotype.Service
+
+@Service
+class PokemonService(private val pokemonRepository: PokemonRepository) {
+
+    fun getAll(): List<Pokemon> {
+        return pokemonRepository.findAll()
+    }
+
+    fun getByPokedex(pokedex: Long): Pokemon {
+        val pokemon = pokemonRepository.findByPokedex(pokedex)
+        return pokemon ?: throw PokemonNotFoundException("Pokemon with Pokedex $pokedex not found")
```

Comment:
Nice, but I see the same string repeated every time you construct this exception.
What if you need to change it at some point?

Wouldn't it be nicer if the constructor accepted the pokedex number, and constructed the message itself?